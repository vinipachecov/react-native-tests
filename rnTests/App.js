import React, { Component } from 'react';
import {
  ScrollView,
  StyleSheet,
  View,
} from 'react-native';

import { theme } from './src/theme';
import Header from './src/components/Header';
import ButtonToggleModal from './src/components/ButtonToggleModal';
import ModalAddBill from './src/components/ModalAddBill';
import Bill from './src/components/Bill';

class App extends Component {
  state = {
    bills: [],
    error: null,
    modalOpen: false,
  };

  addBill = bill => {
    const { bills } = this.state;
    if (!bill.label || !bill.dueDay) return;
    const newBills = [...bills, bill];
    this.setState({ bills: newBills, modalOpen: false });
  }

  render() {
    const { bills, modalOpen } = this.state;

    return (
      <View style={styles.container}>
        <Header />
        <ButtonToggleModal toggleModal={() => this.setState({ modalOpen: true })} />
        <ScrollView>
          {bills.map((bill, index) => (
            <Bill
              key={index.toString()}
              bill={bill}
            />
          ))}
        </ScrollView>
        <ModalAddBill
          addBill={this.addBill}
          modalVisible={modalOpen}
        />
      </View>
    );
  }
}

export default App;

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.colors.grey.light,
    flex: 1,
  },
});
