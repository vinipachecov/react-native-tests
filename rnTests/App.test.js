import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';

describe('[Component] App', () => {
  const tree = create(<App />)

  it('Matches default snapshot', () => {
    expect(tree.toJSON()).toMatchSnapshot();    
  });


  it('Matches initial state', () => {
    expect(tree.root.instance.state).toEqual({ bills: [], error: null, modalOpen: false });    
  });

  it('AddBill updates bills in state', () => {
    console.log(tree.root.findByProps('addBill'));
  });
  
})
