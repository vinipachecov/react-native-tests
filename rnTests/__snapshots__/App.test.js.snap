// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] App Matches default snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#0d47a1",
        "display": "flex",
        "height": 128,
        "justifyContent": "flex-end",
        "padding": 8,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 36,
          "fontWeight": "bold",
          "marginTop": 4,
        }
      }
    >
      My bills
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#5472d3",
        "opacity": 1,
        "padding": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      Create new bill
    </Text>
  </View>
  <RCTScrollView>
    <View />
  </RCTScrollView>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Bill description (rent, credit card, ...)"
        style={
          Object {
            "height": 50,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        keyboard="numeric"
        onChangeText={[Function]}
        placeholder="Bill due day (08, ...)"
        style={
          Object {
            "height": 50,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        keyboard="numeric"
        onChangeText={[Function]}
        placeholder="Bill amount (1500, ...)"
        style={
          Object {
            "height": 50,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Add bill
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
